### PR TITLE
[4.0] Array to php 5.4+ notation [Plugins 3: FileSystem, Finder, Installer, Media-Action, QuickIcon and Search]

### PR DESCRIPTION
**Pull request by andrepereiradasilva**
Pull Request for Improvement.

### Summary of Changes

Since 4.0 doesn't support php 5.3 anymore, use php 5.4+ array notation in FileSystem, Finder, Installer, Media-Action, QuickIcon and Search plugins.

### Testing Instructions

Simple code review.

### Documentation Changes Required

None.

### CODE REVIEW

**Comment by frankmayer** [comment]
I have tested this item :white_check_mark: successfully on 181d79f457cfbeff83fdf150c5127e9369c735cb<br /><br />Code review.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/16654">issues.joomla.org/tracker/joomla-cms/16654</a>.</sub>

File: plugins/search/categories/categories.php [hunk 1]
```diff
@@ -69,14 +69,14 @@ public function onContentSearch($text, $phrase = '', $ordering = '', $areas = nu
...
-				return array();
+				return [];
 			}
 		}
 
 		$sContent = $this->params->get('search_content', 1);
 		$sArchived = $this->params->get('search_archived', 1);
 		$limit = $this->params->def('search_limit', 50);
-		$state = array();
+		$state = [];
```
Comment (Quy) [inline plugins/search/categories/categories.php]: Align

File: plugins/search/content/content.php [hunk 2]
```diff
@@ -143,7 +143,7 @@ public function onContentSearch($text, $phrase = '', $ordering = '', $areas = nu
 				break;
 		}
 
-		$rows = array();
+		$rows = [];
```
Comment (Quy) [inline plugins/search/content/content.php]: Align

File: plugins/search/newsfeeds/newsfeeds.php [hunk 3]
```diff
@@ -66,14 +66,14 @@ public function onContentSearch($text, $phrase = '', $ordering = '', $areas = nu
...
-				return array();
+				return [];
 			}
 		}
 
 		$sContent = $this->params->get('search_content', 1);
 		$sArchived = $this->params->get('search_archived', 1);
 		$limit = $this->params->def('search_limit', 50);
-		$state = array();
+		$state = [];
```
Comment (Quy) [inline plugins/search/newsfeeds/newsfeeds.php]: Align

File: plugins/search/newsfeeds/newsfeeds.php [hunk 4]
```diff
@@ -87,21 +87,21 @@ public function onContentSearch($text, $phrase = '', $ordering = '', $areas = nu
...
-			return array();
+			return [];
 		}
 
 		switch ($phrase)
 		{
 			case 'exact':
 				$text = $db->quote('%' . $db->escape($text, true) . '%', false);
-				$wheres2 = array();
+				$wheres2 = [];
```
Comment (Quy) [inline plugins/search/newsfeeds/newsfeeds.php]: Align

File: plugins/search/newsfeeds/newsfeeds.php [hunk 5]
```diff
@@ -111,12 +111,12 @@ public function onContentSearch($text, $phrase = '', $ordering = '', $areas = nu
...
 			default:
 				$words = explode(' ', $text);
-				$wheres = array();
+				$wheres = [];
 
 				foreach ($words as $word)
 				{
 					$word = $db->quote('%' . $db->escape($word, true) . '%', false);
-					$wheres2 = array();
+					$wheres2 = [];
```
Comment (Quy) [inline plugins/search/newsfeeds/newsfeeds.php]: Align

File: plugins/search/newsfeeds/newsfeeds.php [hunk 6]
```diff
@@ -111,12 +111,12 @@ public function onContentSearch($text, $phrase = '', $ordering = '', $areas = nu
 			case 'any':
 			default:
 				$words = explode(' ', $text);
-				$wheres = array();
+				$wheres = [];
```
Comment (Quy) [inline plugins/search/newsfeeds/newsfeeds.php]: Align

File: plugins/search/categories/categories.php (hunk 1 above)
Comment (andrepereiradasilva) [inline plugins/search/categories/categories.php]: all aligned

**Pull request closed without merging**